nsxt/resource_nsxt_policy_edge_transport_node.go

```diff
@@ -0,0 +1,1669 @@
+	model.SyslogConfiguration_LOG_LEVEL_ERROR,
+	model.SyslogConfiguration_LOG_LEVEL_WARNING,
+	model.SyslogConfiguration_LOG_LEVEL_NOTICE,
+	model.SyslogConfiguration_LOG_LEVEL_INFO,
+	model.SyslogConfiguration_LOG_LEVEL_DEBUG,
+}
+
+var policySyslogProtocolValues = []string{
+	model.SyslogConfiguration_PROTOCOL_TCP,
+	model.SyslogConfiguration_PROTOCOL_UDP,
+	model.SyslogConfiguration_PROTOCOL_TLS,
+	model.SyslogConfiguration_PROTOCOL_LI,
+	model.SyslogConfiguration_PROTOCOL_LI_TLS,
+}
+
+var policyCpuReservationValues = []string{
+	model.CPUReservation_RESERVATION_IN_SHARES_EXTRA_HIGH_PRIORITY,
+	model.CPUReservation_RESERVATION_IN_SHARES_HIGH_PRIORITY,
+	model.CPUReservation_RESERVATION_IN_SHARES_NORMAL_PRIORITY,
+	model.CPUReservation_RESERVATION_IN_SHARES_LOW_PRIORITY,
+}
+
+var managementAssignments = []string{
+	"dhcp_v4",
+	"static_ipv4",
+	"static_ipv6",
+}
+
+var tepAssignments = []string{
+	"dhcp_v4",
+	"dhcp_v6",
+	"static_ipv4_list",
+	"static_ipv4_pool",
+	"static_ipv6_list",
+	"static_ipv6_pool",
+}
+
+func getPolicyIPAssignmentSchema(required bool, minItems, maxItems int, validAssignments []string) *schema.Schema {
+	allAssignments := map[string]*schema.Schema{
+		"auto_conf": {
+			Type:        schema.TypeBool,
+			Optional:    true,
+			Description: "Enable Auto-conf based IPv6 assignment",
+		},
+		"dhcp_v4": {
+			Type:        schema.TypeBool,
+			Optional:    true,
+			Description: "Enable DHCP based IPv4 assignment",
+		},
+		"dhcp_v6": {
+			Type:        schema.TypeBool,
+			Optional:    true,
+			Description: "Enable DHCP based IPv6 assignment",
+		},
+		"no_assignment": {
+			Type:        schema.TypeBool,
+			Optional:    true,
+			Description: "Enable no IP assignment",
+		},
+		"static_ipv4": {
+			Type:        schema.TypeList,
+			Description: "IP assignment specification for a Static IP",
+			Optional:    true,
+			MaxItems:    1,
+			Elem: &schema.Resource{
+				Schema: map[string]*schema.Schema{
+					"default_gateway": {
+						Type:         schema.TypeString,
+						Required:     true,
+						Description:  "Default IPv4 gateway for the node",
+						ValidateFunc: validateSingleIP(),
+					},
+					"management_port_subnet": {
+						Type:        schema.TypeList,
+						Required:    true,
+						MaxItems:    1,
+						MinItems:    1,
+						Description: "IPv4 Port subnet for management port",
+						Elem: &schema.Resource{
+							Schema: map[string]*schema.Schema{
+								"ip_addresses": {
+									Type:        schema.TypeList,
+									Required:    true,
+									Description: "IPv4 Addresses",
+									MinItems:    1,
+									MaxItems:    1,
+									Elem: &schema.Schema{
+										Type:         schema.TypeString,
+										ValidateFunc: validateSingleIP(),
+									},
+								},
+								"prefix_length": {
+									Type:         schema.TypeInt,
+									Required:     true,
+									Description:  "Subnet Prefix Length",
+									ValidateFunc: validation.IntBetween(1, 64),
```

Comment:
this is v4 only so I think this should be `IntBetween(1, 32)`
we also have validator for V4 only addresses

Reply:
Makes sense - IDK why in the API spec they chose 64.